code/parse/sexp.cpp

```diff
@@ -3585,6 +3620,33 @@ int get_sexp()
 
 		}
 
+		// Sexp container
+		else if (*Mp == sexp_container::DELIM) {
+			Mp++;
+
+			char container_name[TOKEN_LENGTH];
+			const SCP_string container_delim(1, sexp_container::DELIM);
+			stuff_string(container_name, F_NAME, TOKEN_LENGTH, container_delim.c_str());
+
+			// bump past closing '&'
+			Mp += 2;
+
+			if (get_sexp_container(container_name) == nullptr) {
+				Error(LOCATION, "Attempt to use unknown container '%s'", container_name);
+				return -1;
+			}
+
+			// advance to the control options, since we'll read them when calling get_sexp() below
+			while (*Mp != '(') {
```

Comment:
If the string is malformed, then the loop could iterate way too far.  Add a sanity error-check within the loop (maybe check for `\n` or `\0`).

Reply:
Tried to address this.